src/transforms/remap.rs

```diff
@@ -183,50 +187,60 @@ impl TransformConfig for RemapConfig {
         Input::all()
     }
 
-    fn outputs(&self, merged_definition: &schema::Definition) -> Vec<Output> {
+    fn outputs(&self, input_definition: &schema::Definition) -> Vec<Output> {
         // We need to compile the VRL program in order to know the schema definition output of this
         // transform. We ignore any compilation errors, as those are caught by the transform build
         // step.
         let default_definition = self
             .compile_vrl_program(
                 enrichment::TableRegistry::default(),
-                merged_definition.clone(),
+                input_definition.clone(),
             )
-            .ok()
-            .and_then(|(_, _, _, state)| {
+            .map(|(_, _, _, state)| {
                 let meaning = state
                     .get_external_context::<MeaningList>()
                     .cloned()
                     .expect("context exists")
                     .0;
 
-                state
-                    .target_kind()
-                    .clone()
-                    .into_object()
-                    .map(schema::Definition::from)
-                    .map(|mut def| {
-                        for (id, path) in meaning {
-                            def = def.with_known_meaning(path, &id);
-                        }
-                        def
-                    })
+                let mut new_type_def = Definition::empty_with_kind(
+                    state.target_kind().clone(),
+                    input_definition.log_namespaces().clone(),
+                );
+                for (id, path) in meaning {
+                    new_type_def = new_type_def.with_meaning(path, &id);
+                }
+                new_type_def
             })
-            .unwrap_or_else(schema::Definition::empty);
+            .unwrap_or_else(|_| {
+                Definition::empty_with_kind(
+                    Kind::never(),
```

Comment:
I'll add a comment, but it's pretty much what you assumed. This type information isn't used, but it matches the closest to the actual value. You can read the type quite literally as "the program can _never_ return a value" (since it failed to compile).

This isn't the case here, but there are other places where it's used since it has the nice property of becoming whatever it's merged with, so for example the `merged_definition` function defaults to `never` and merges all the possible definitions into that.